### ISSUE TITLE
No Labels Found

### ISSUE THREAD

**Issue by yacqubstack** [ultralytics/ultralytics#18791]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

<img width="160" alt="Image" src="https://github.com/user-attachments/assets/7a3bcedd-9ea3-4033-8b89-2223dc0886bd" />

this is my coco dataset structure and im trying to train it but whenever i put this in the terminal:
yolo segment train
data=coco.yaml
model=yolov8n-seg.pt
epochs=20
batch=4

i get this error?
train: Scanning /content/drive/Documents/app
train: WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/train.cache. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
train: WARNING ⚠️ Cache directory /content/drive/Documents/app/coco/labels is not writeable, cache not saved.
WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/train.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
val: Scanning /content/drive/Documents/app/c
val: WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/val.cache. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
val: WARNING ⚠️ Cache directory /content/drive/Documents/app/coco/labels is not writeable, cache not saved.
WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/val.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.

this is my coco.yaml file: 
path: /content/drive/Documents/app/coco
train: images/train
val: images/val
nc: 46

I've done Python checks and it confirms:

Every file_name in your COCO "images" array matches an actual image file,
Every annotation has a valid image_id and category_id,
Every annotation has either bbox or segmentation.

but it still doesnt work?

### Environment

train: Scanning /content/drive/Documents/app
train: WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/train.cache. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
train: WARNING ⚠️ Cache directory /content/drive/Documents/app/coco/labels is not writeable, cache not saved.
WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/train.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
val: Scanning /content/drive/Documents/app/c
val: WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/val.cache. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
val: WARNING ⚠️ Cache directory /content/drive/Documents/app/coco/labels is not writeable, cache not saved.
WARNING ⚠️ No labels found in /content/drive/Documents/app/coco/labels/val.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.

### Minimal Reproducible Example

this is my coco.yaml file: 
path: /content/drive/Documents/app/coco
train: images/train
val: images/val
nc: 46

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yacqubstack, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) for assistance with dataset setup, where many [best practices](https://docs.ultralytics.com/datasets/) and common issues regarding dataset formatting are addressed.

From the issue details you provided, the warning indicates that no labels were found, which suggests an issue with your dataset structure or annotations. To resolve this, please ensure that:
1. Your `coco/labels` directory contains valid `.txt` files for each corresponding image in `coco/images/train` and `coco/images/val`, as outlined in the [Dataset Documentation](https://docs.ultralytics.com/datasets/).
2. The class labels and annotations in these `.txt` files conform to the YOLO format.

If this is a 🐛 Bug Report, could you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) showcasing your issue? This should include:
- Your `coco.yaml` file (shared above, thank you!)
- A small sample of your dataset directory structure, ensuring the inclusion of both `images` and `labels` subdirectories
- A few example `.txt` label files matching your images

If the issue persists, please also try running the following command to ensure you’re using the latest version of the `ultralytics` package:

```bash
pip install -U ultralytics
```

Additionally, ensure your environment meets all [necessary requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), including Python>=3.8 and PyTorch>=1.8.

For interactive community support, join us on [Discord](https://discord.com/invite/ultralytics) 🎧. You can also find discussions on [Discourse](https://community.ultralytics.com/) and our [Subreddit](https://reddit.com/r/Ultralytics) to engage with others working on similar tasks.

## Upgrade

Please ensure you're using the latest `ultralytics` version for any bug fixes:
```bash
pip install -U ultralytics
```

## Environments

YOLO may also be executed in one of the pre-verified environments for smoother execution and debugging. Try it for free:
- **Notebooks** with GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Rest assured, an Ultralytics engineer will review your issue and assist you further soon. Thank you for your patience and for using YOLO! 😊

**Comment by Y-T-G** [comment]
Why is it called `coco.yaml`? Can you try a different name? `coco.yaml` is a built-in yaml file to train on the COCO dataset.

**Comment by Y-T-G** [comment]
Also labels are supposed to be in YOLO format, not COCO format like in your screenshot.

https://docs.ultralytics.com/datasets/detect/#ultralytics-yolo-format

**Comment by yacqubstack** [comment]
Hi @Y-T-G ,

Thank you for your input and for pointing me in the right direction!

I wanted to clarify that I’m working on an image segmentation task using the Fashionpedia dataset: https://github.com/cvdfoundation/fashionpedia rather than object detection. Fashionpedia provides detailed segmentation masks, which is why my dataset is structured with masks instead of labels. Here’s a brief overview of my current setup:
Dataset Structure:

<img width="167" alt="Image" src="https://github.com/user-attachments/assets/26d25ddf-42eb-4db4-9dcc-c4eee5a4f839" />



Training Command:
yolo segment train data=data.yaml model=yolov8n-seg.pt epochs=100 imgsz=640 

data.yaml Configuration:
path: /content/drive/Documents/app/coco

train: images/train
val: images/val

masks:
  train: masks/train
  val: masks/val

nc: 46

names:
  0: Dress
  1: Top
  2: Skirt
  3: Pants
  4: Shorts
  5: Jacket
  6: Coat
  7: Sweater
  8: Blouse
  9: Shirt
  10: Jeans
  11: Hoodie
  12: T-shirt
  13: Tank Top
  14: Leggings
  15: Polo Shirt
  16: Cardigan
  17: Blazer
  18: Sportswear
  19: Jumpsuit
  20: Romper
  21: Socks
  22: Shoes
  23: Sneakers
  24: Boots
  25: Sandals
  26: Heels
  27: Flats
  28: Belt
  29: Hat
  30: Scarf
  31: Gloves
  32: Necklace
  33: Earrings
  34: Bracelet
  35: Watch
  36: Sunglasses
  37: Tie
  38: Bow-tie
  39: Vest
  40: Underwear
  41: Swimsuit
  42: Raincoat
  43: Purse
  44: Backpack
  45: Wallet

Given that segmentation requires mask files instead of label .txt files, I’ve set up my data.yaml accordingly and am using the segment mode with the appropriate model (yolov8n-seg.pt).

Current Issue: Despite these configurations, YOLOv8 is still attempting to access a labels directory, which doesn't exist in my dataset structure. I’ve also ensured that all mask files are in .png format and correctly named to match their corresponding images.

Steps I’ve Taken So Far:

Verified data.yaml: Ensured there are no references to labels.
Cleared Cache Files: Removed any residual cache files that might reference labels.
Checked Mask Integrity: Confirmed that all masks are properly formatted and correspond to their images.

Request: Could you please provide guidance or suggest any additional steps I might take to ensure YOLOv8 recognizes the segmentation setup and stops searching for the labels directory?

Thank you for your assistance!

Best regards, @yacqubstack

**Comment by Skillnoob** [comment]
Use this instead for segmentation:
https://docs.ultralytics.com/datasets/segment/#coco-dataset-format-to-yolo-format
